### PR TITLE
refactor: remove unused processor error variants

### PR DESCRIPTION
**Pull request by vlopes11**
related issue: #774

### CODE REVIEW

**Review by hackaugusto — APPROVED** [review]
LGTM

File: processor/src/errors.rs [hunk 1]
```diff
@@ -16,11 +16,9 @@ use std::error::Error;
 pub enum ExecutionError {
     AdviceKeyNotFound(Word),
     AdviceStackReadFailed(u32),
-    MerkleNodeIndex(Felt, Felt),
+    MerkleNodeIndex { depth: Felt, value: Felt },
     MerkleUpdateInPlace,
-    MerkleSetLookupFailed(MerkleError),
-    MerkleSetNotFound([u8; 32]),
-    MerkleSetUpdateFailed(MerkleError),
+    MerkleStoreLookupFailed(MerkleError),
```
Comment (bobbinth) [inline processor/src/errors.rs:21]: I'm thinking we should make the following changes here:

* `MerkleNodeIndex` rename to `InvalidNodeIndex`.
* Add `MerkleStoreUpdateFailed`.

File: processor/src/advice/mem_provider.rs [hunk 2]
```diff
@@ -119,12 +129,16 @@ impl AdviceProvider for MemAdviceProvider {
...
+            ExecutionError::MerkleNodeIndex {
+                depth: *depth,
+                value: *index,
+            }
+        })?;
         self.store
             .set_node(root, node_index, value)
             .map(|root| root.path)
-            .map_err(ExecutionError::MerkleSetLookupFailed)
+            .map_err(ExecutionError::MerkleStoreLookupFailed)
```
Comment (bobbinth) [inline processor/src/advice/mem_provider.rs:141]: We should probably change this to `MerkleStoreUpdateFailed`.

File: processor/src/errors.rs [hunk 3]
```diff
@@ -50,11 +48,9 @@ impl Display for ExecutionError {
                 let hex = to_hex(Felt::elements_as_bytes(key))?;
                 write!(fmt, "Can't push values onto the advice stack: value for key {hex} not present in the advice map.")
             }
-            MerkleNodeIndex(depth, index) => write!(fmt, "The provided depth {depth} cannot be represented as Merkle index with the value {index}"),
+            MerkleNodeIndex { depth, value } => write!(fmt, "The provided depth {depth} cannot be represented as Merkle index with the value {value}"),
```
Comment (bobbinth) [inline processor/src/errors.rs:51]: I would update message to something like: "The specified index {index} is invalid for the specified depth {depth}"

**Review by bobbinth — CHANGES_REQUESTED** [review]
Looks good! Thank you! I left a couple of minor comments inline.

File: processor/src/errors.rs (hunk 1 above)
Comment (vlopes11) [reply processor/src/errors.rs]: Probably was outdated; this was done on the previous commit

File: processor/src/advice/mem_provider.rs (hunk 2 above)
Comment (vlopes11) [reply processor/src/advice/mem_provider.rs:141]: Done!

Comment (bobbinth) [reply processor/src/advice/mem_provider.rs:141]: I think this is still `MerkleStoreLookupFailed` - it should be `MerkleStoreUpdateFailed`.

**Review by bobbinth — APPROVED** [review]
All looks good! Thank you!

**Pull request merged**